### PR TITLE
CONSOLE-2494: Bump node and yarn versions in docker builder file

### PR DESCRIPTION
**Pull request by zherman0**
Need to wait for [ART-2456] to be complete and for PR7304 to merge.

This bumps dependencies for nodejs and yarn in the build file.

### CODE REVIEW

**Comment by zherman0** [comment]
/hold

Waiting for PR7304 to merge

**Comment by spadgett** [comment]
> Need to wait for [ART-2456] to be complete and for PR7304 to merge.
> 
> This bumps dependencies for nodejs and yarn in the build file.

@zherman0 We should combine this with #7221 so we only update the builder image once. cc @suomiy

**Comment by spadgett** [comment]
Let's update kubectl as well.

**Comment by suomiy** [comment]
> @zherman0 We should combine this with #7221 so we only update the builder image once. cc @suomiy

+1 feel free to move the changes from https://github.com/openshift/console/pull/7221 to this PR

**Comment by spadgett** [comment]
I don't think we need to wait for the ART change although #7304 is in the merge queue.

**Comment by spadgett** [comment]
@suomiy I added your commit from #7221 to @zherman0's PR. I'm about to build and push the image to Quay.

**Comment by suomiy** [comment]
@spadgett 
great! Let me know once you have the image so I can rerun the tests in https://github.com/openshift/release/pull/13715

**Comment by yselkowitz** [comment]
Wouldn't the yarn tarball name in Dockerfile.product also need to be updated for this?

**Comment by spadgett** [comment]
> Wouldn't the yarn tarball name in Dockerfile.product also need to be updated for this?

Isn't that part of #7304? I don't think it needs to be in one PR. It shouldn't break anything if done separately.

**Comment by yselkowitz** [comment]
#7304 doesn't touch yarn, only nodejs.

**Comment by suomiy** [comment]
it seems that the new node/yarn may have broken the analyze script: https://github.com/openshift/release/pull/13715

**Comment by zherman0** [comment]
@spadgett - I think all the other dependencies for this update has been merged.  Can we get an LGTM?

**Comment by spadgett** [comment]
@suomiy Sorry, missed this. It looks like we're running into this problem? https://github.com/webpack/webpack/issues/4629

**Comment by zherman0** [comment]
/hold
Holding for the PR from openshift

**Comment by suomiy** [comment]
> @suomiy Sorry, missed this. It looks like we're running into this problem? [webpack/webpack#4629](https://github.com/webpack/webpack/issues/4629)

@spadgett seems like this issue got fixed recently. Should we update webpack-cli to the newest version to proceed with this one?

**Comment by spadgett** [comment]
> > @suomiy Sorry, missed this. It looks like we're running into this problem? [webpack/webpack#4629](https://github.com/webpack/webpack/issues/4629)
> 
> @spadgett seems like this issue got fixed recently. Should we update webpack-cli to the newest version to proceed with this one?

Sounds good to me. We're working on updating our dependencies anyway. Thanks!

**Comment by suomiy** [comment]
@spadgett @zherman0 Is there an update on this one? I also saw the expect library has been removed from this PR, can you please elaborate on the reason?

**Comment by spadgett** [comment]
@zherman0 It looks like you lost the changes from commit 18f12ad2f9fef4b9d2e232df3d6b6e0d03526781 in the last push.

**Comment by spadgett** [comment]
@zherman0 We'll also need to update webpack-cli to move ahead with this change, otherwise the analyze script is broken.

**Comment by spadgett** [comment]
The webpack-cli change will need to be a separate PR that we can merge first. We need to the rehearse jobs to pass on the release PR before we merge this Dockerfile change.

**Comment by suomiy** [comment]
should I make the PR? And do we want to bump tectonic-console-builder version for that, or rewrite v22?

**Comment by spadgett** [comment]
> should I make the PR? And do we want to bump tectonic-console-builder version for that, or rewrite v22?

@zherman0 Was going to look at that.

I don't think we need a builder image change to update webpack-cli.

**Comment by spadgett** [comment]
/override ci/prow/ceph-storage-plugin

The ceph-storage-plugin job should only have run on PRs with changes to `frontend/packages/ceph-storage-plugin`. It failed due to an unrelated CI config change. Overriding status on this PR.

**Comment by spadgett** [comment]
@christianvogt @vojtechszocs FYI, this PR bumps node. We need to bump webpack-cli first, though.

**Comment by vojtechszocs** [comment]
@spadgett @zherman0 For sake of consistency, we should also update `Dockerfile.plugins.demo` which uses the Console builder image.

**Comment by bparees** [comment]
> @bparees can you lgtm

i think that should come from @spadgett on this one.

/assign @spadgett

File: Dockerfile.builder
Comment (spadgett) [inline Dockerfile.builder:20]: Let's go ahead and use 

```suggestion
ENV NODE_VERSION="v14.16.0"
```

**Review by spadgett — COMMENTED** [review]
@zherman0 I think we should go ahead and move to the latest kubectl as well, 1.20.4.

File: Dockerfile.builder
Comment (spadgett) [reply Dockerfile.builder:20]: Actuallly, 14.15.4 might be better to align with what ART will use, and 14.15 is LTS

```suggestion
ENV NODE_VERSION="v14.15.4"
```

Comment (yselkowitz) [reply Dockerfile.builder:20]: It turns out RHEL 8 just bumped to 14.16.0 for several CVEs, so we probably need to retarget that.

Comment (spadgett) [reply Dockerfile.builder:20]: I'm good with 14.16.0.

**Comment by spadgett** [comment]
@zherman0 Can you update the `engines` value in package.json and also update the README for the new node version?

**Comment by spadgett** [comment]
openshift/release#13715 has merged

**Comment by spadgett** [comment]
I just built and pushed the latest image to quay: https://quay.io/repository/coreos/tectonic-console-builder?tab=tags

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**
